src/transform.ts

```diff
@@ -1,33 +1,29 @@
 import {Filter} from './filter';
 
-/**
- * Top-level transform object.
- */
-export interface Transform {
+export interface FilterTransform {
   /**
    * A string containing the filter Vega expression. Use `datum` to refer to the current data object.
    */
-  filter?: string | Filter | (string | Filter)[];
-  /**
-   * Whether to filter invalid values (`null` and `NaN`) from the data. By default (`undefined`), only quantitative and temporal fields are filtered. If set to `true`, all data items with null values are filtered. If `false`, all data items are included.
-   */
-  filterInvalid?: boolean;
-  /**
-   * Calculate new field(s) using the provided expresssion(s). Calculation are applied before filter.
-   */
-  calculate?: Formula[];
+  filter: Filter | Filter[];
```

Comment:
I thought you wanna get rid of `Filter[]`.  (This could become a separate PR though.)